dbg.h

```diff
@@ -312,6 +314,38 @@ using std::size;
 #endif
 }  // namespace
 
+// Specializations for container adapters
+
+template <class T, class C>
+T pop(std::stack<T, C>& adapter) {
+  T value = std::move(adapter.top());
```

Comment:
That `std::move` seems unnecessary? `adapter.front()` is an r-value already. Similar below.

Reply:
Are you sure? Because std::stack::top seems to returning a reference. A quick check shows without `std::move` an unnecessary copy being incurred.
```cpp
#include <iostream>
#include <stack>

struct MyStruct {
  MyStruct() = default;

  MyStruct(const MyStruct&) noexcept {
    std::cerr << "copied" << std::endl;
  }

  MyStruct(MyStruct&&) noexcept {
    std::cerr << "moved" << std::endl;
  }
};

int main() {
  std::stack<MyStruct> st;
  st.emplace(); // push element, create inplace
  std::cerr << st.size() << std::endl;

  MyStruct b = st.top(); // element is copied
  // MyStruct c = std::move(st.top()); // element is moved
  st.pop();
}

```

Reply:
You are right, my bad.